code-challenges.test.js

```diff
@@ -15,37 +15,129 @@
+    //Note: this process works backwards starting with the last index
+  }
+  //return the shifted and shuffled array
+  return array
+}
+//Example Input: ["purple", "blue", "green", "yellow", "pink"]
+//Example Output: ["pink", "blue", "green", "yellow"]
+
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+describe('voteCount', () => {
+  it('takes in an object that contains up votes and down votes and returns the end tally', () => {
+
+    const votes1 = { upVotes: 13, downVotes: 2 }
+    // Expected output: 11
+    const votes2 = { upVotes: 2, downVotes: 33 }
+    // Expected output: -31
+
+    expect(voteCount(votes1)).toEqual(11)
+    expect(voteCount(votes2)).toEqual(-31)
+  })
+})
+
 
 // b) Create the function that makes the test pass.
 
+//create a function named voteCount that takes an oobject as a parameter
+const voteCount = (object) => {
+  //use dot notation to access the keys values (which are numbers) and subtract them
+  //return the number
+  return object.upVotes - object.downVotes
+}
+
+//Example Input: {upVotes: 2, downVotes: 100}
+//Example Output: -98
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
+describe('noDupes', () => {
+  it('takes in two arrays as arguments and returns one array with no duplicate values', () => {
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+
+    expect(noDupes(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
 // b) Create the function that makes the test pass.
+
+//create a function named 'noDupes' that takes 2 arrays as parameters
+const noDupes = (array1, array2) => {
+  //use the .concat() method to combine the two arrays
+  //use the .filter() method to isolate the value, index, and array
+  //while iterating through the array, filter out values that already exist
+  //return that new array
+  return array1.concat(array2).filter((value, index, array) => array.indexOf(value) === index)
+}
+
```

Comment:
Nice use of `.filter()` and comparing the index of the first occurrence of the value to the index used in the filter method! Great work!